src/pudl/analysis/record_linkage/embed_dataframe.py

```diff
@@ -32,6 +35,7 @@ class FeatureMatrix:
     """
 
     matrix: np.ndarray | scipy.sparse.csr_matrix
+    index: pd.Index
```

Comment:
I added this variable to keep track of the index of the matrix, because in the FERC to EIA context, it's useful to know what `record_id_eia` x `record_id_ferc1` pair each row in the matrix corresponds to.